packages/flutter/test/painting/image_cache_test.dart

```diff
@@ -9,78 +9,117 @@ import '../rendering/rendering_tester.dart';
 
-    // cache has three entries: 3(9), 4(11), 1(1)
+      // cache has three entries: 3(9), 4(11), 1(1)
 
-    final TestImageInfo m = await extractOneFrame(const TestImageProvider(2, 13).resolve(ImageConfiguration.empty));
-    expect(m.value, equals(13));
+      final TestImageInfo m = await extractOneFrame(const TestImageProvider(2, 13).resolve(ImageConfiguration.empty));
+      expect(m.value, equals(13));
 
-    // cache has three entries: 4(11), 1(1), 2(13)
+      // cache has three entries: 4(11), 1(1), 2(13)
 
-    final TestImageInfo n = await extractOneFrame(const TestImageProvider(3, 14).resolve(ImageConfiguration.empty));
-    expect(n.value, equals(14));
+      final TestImageInfo n = await extractOneFrame(const TestImageProvider(3, 14).resolve(ImageConfiguration.empty));
+      expect(n.value, equals(14));
 
-    // cache has three entries: 1(1), 2(13), 3(14)
+      // cache has three entries: 1(1), 2(13), 3(14)
 
-    final TestImageInfo o = await extractOneFrame(const TestImageProvider(4, 15).resolve(ImageConfiguration.empty));
-    expect(o.value, equals(15));
+      final TestImageInfo o = await extractOneFrame(const TestImageProvider(4, 15).resolve(ImageConfiguration.empty));
+      expect(o.value, equals(15));
 
-    // cache has three entries: 2(13), 3(14), 4(15)
+      // cache has three entries: 2(13), 3(14), 4(15)
 
-    final TestImageInfo p = await extractOneFrame(const TestImageProvider(1, 16).resolve(ImageConfiguration.empty));
-    expect(p.value, equals(16));
+      final TestImageInfo p = await extractOneFrame(const TestImageProvider(1, 16).resolve(ImageConfiguration.empty));
+      expect(p.value, equals(16));
 
-    // cache has three entries: 3(14), 4(15), 1(16)
+      // cache has three entries: 3(14), 4(15), 1(16)
 
+    });
+
+    test('clear removes all images and resets cache size', () async {
+      const TestImage testImage = const TestImage(width: 8, height: 8);
+
+      expect(imageCache.currentSize, 0);
+      expect(imageCache.currentSizeBytes, 0);
+
+      await extractOneFrame(const TestImageProvider(1, 1, image: testImage).resolve(ImageConfiguration.empty));
+      await extractOneFrame(const TestImageProvider(2, 2, image: testImage).resolve(ImageConfiguration.empty));
+
+      expect(imageCache.currentSize, 2);
+      expect(imageCache.currentSizeBytes, 256 * 2);
+
+      imageCache.clear();
+
+      expect(imageCache.currentSize, 0);
+      expect(imageCache.currentSizeBytes, 0);
+    });
+
+    test('evicts individual images', () async {
+      const TestImage testImage = const TestImage(width: 8, height: 8);
+      await extractOneFrame(const TestImageProvider(1, 1, image: testImage).resolve(ImageConfiguration.empty));
+      await extractOneFrame(const TestImageProvider(2, 2, image: testImage).resolve(ImageConfiguration.empty));
+
+      expect(imageCache.currentSize, 2);
+      expect(imageCache.currentSizeBytes, 256 * 2);
+      expect(imageCache.evict(1), true);
```

Comment:
Can we test that this works with passing the same ImageProvider that you pass to Image to actually load the image, the way that the docs for evict say should work?

Reply:
tester here: https://github.com/flutter/flutter/pull/18135/files#diff-221f0251a33dbe10c87cd33eb1a8ab51R27

Reply:
excellent